graviti/utility/lazy.py

```diff
@@ -0,0 +1,242 @@
+        return self._parent.pages[self._pos].__getitem__(index)  # type: ignore[no-any-return]
+
+    def __iter__(self) -> Iterator[_T]:
+        self._fetcher(self._pos)
+        return self._parent.pages[self._pos].__iter__()  # type: ignore[no-any-return]
+
+
+class LazyFactory:
+    """LazyFactory is a factory for creating lazy lists.
+
+    Arguments:
+        total_count: The total count of the elements in the lazy lists.
+        limit: The size of each lazy load page.
+        getter: A callable object to get the source data.
+
+    Examples:
+        >>> TOTAL_COUNT = 1000
+        >>> def getter(offset: int, limit: int) -> Dict[str, Any]:
+        ...     stop = min(offset + limit, TOTAL_COUNT)
+        ...     data = [
+        ...         {
+        ...             "remotePath": f"{i:06}.jpg",
+        ...             "label": {"CLASSIFICATION": {"category": "cat" if i % 2 else "dog"}},
+        ...         }
+        ...         for i in range(offset, stop)
+        ...     ]
+        ...
+        ...     return {
+        ...         "data": data,
+        ...         "offset": offset,
+        ...         "recordSize": len(data),
+        ...         "totalCount": TOTAL_COUNT,
+        ...     }
+        >>> factory = LazyFactory(TOTAL_COUNT, 128, getter)
+        >>> paths = factory.create_list(
+        ...     lambda data: (item["remotePath"] for item in data["data"]), dtype="<U10"
+        ... )
+        >>> categories = factory.create_list(
+        ...     lambda data: (item["label"]["CLASSIFICATION"]["category"] for item in data["data"]),
+        ...     dtype="<U3",
+        ... )
+        >>> paths
+        LazyList [
+          '000000.jpg',
+          '000001.jpg',
+          '000002.jpg',
+          '000003.jpg',
+          '000004.jpg',
+          '000005.jpg',
+          '000006.jpg',
+          '000007.jpg',
+          '000008.jpg',
+          '000009.jpg',
+          '000010.jpg',
+          '000011.jpg',
+          '000012.jpg',
+          '000013.jpg',
+          ... (985 items are folded),
+          '000999.jpg'
+        ]
+        >>> categories
+        LazyList [
+          'dog',
+          'cat',
+          'dog',
+          'cat',
+          'dog',
+          'cat',
+          'dog',
+          'cat',
+          'dog',
+          'cat',
+          'dog',
+          'cat',
+          'dog',
+          'cat',
+          ... (985 items are folded),
+          'cat'
+        ]
+
+    """
+
+    def __init__(self, total_count: int, limit: int, getter: Callable[[int, int], Any]) -> None:
+        self._getter = getter
+        self._total_count = total_count
+        self._limit = limit
+        self._lists: List[LazyList[Any]] = []
+
+    def create_list(self, extractor: Callable[[Any], Iterable[Any]], dtype: Any) -> LazyList[Any]:
+        """Create a lazy list from the factory.
+
+        Arguments:
+            extractor: A callable object to make the source data to a iterable object.
```

Comment:
`to an iterable object`

Reply:
what about this?

Reply:
I don't understand? can you provide more hints for your comment?